FRAuth/FRAuth/Callbacks/DeviceBindingCallback.swift

```diff
@@ -143,29 +143,35 @@ open class DeviceBindingCallback: MultipleValuesCallback, Binding {
     
     
     /// Bind the device.
-    /// - Parameter completion Completion block for Device binding result callback
-    open func bind(completion: @escaping DeviceBindingResultCallback) {
+    /// - Parameter deviceAuthenticator: method for providing a ``DeviceAuthenticator`` from ``DeviceBindingAuthenticationType`` - provide nil to default to `deviceAuthenticatorIdentifier`
+    /// - Parameter completion: Completion block for Device binding result callback
+    open func bind(deviceAuthenticator: ((DeviceBindingAuthenticationType) -> DeviceAuthenticator)?,
+                   completion: @escaping DeviceBindingResultCallback) {
+        
+        let authInterface = deviceAuthenticator?(deviceBindingAuthenticationType) ?? deviceAuthenticatorIdentifier(deviceBindingAuthenticationType)
         let dispatchQueue = DispatchQueue(label: "com.forgerock.concurrentQueue", qos: .userInitiated)
         dispatchQueue.async {
-            self.execute(authInterface: nil, deviceId: nil, encryptedPreference: nil, completion)
+            self.execute(authInterface: authInterface, deviceId: nil, deviceRepository: nil, completion)
         }
     }
     
     
     /// Helper method to execute binding , signing, show biometric prompt.
-    /// - Parameter authInterface: Interface to find the Authentication Type - provide nil to default to getDeviceBindingAuthenticator()
-    /// - Parameter deviceId: Interface to find the Authentication Type - provide nil to default to FRDevice.currentDevice?.identifier.getIdentifier()
-    /// - Parameter completion: completion Completion block for Device binding result callback
+    /// - Parameter authInterface: Interface to find the Authentication Type - provide nil to default to ``getDeviceAuthenticator(type: deviceBindingAuthenticationType)``
+    /// - Parameter deviceId: Interface to find the Authentication Type - provide nil to default to `FRDevice.currentDevice?.identifier.getIdentifier()`
+    /// - Parameter deviceRepository: Storage for user keys - provide nil to default to ``KeychainDeviceRepository``
+    /// - Parameter completion: Completion block for Device binding result callback
     internal func execute(authInterface: DeviceAuthenticator?,
```

Comment:
Minor suggestion to have default =nil values for the optional parameters of the execute method, to provide a cleaner interface

Reply:
Great suggestion! Made the changes where it applies.